report.go

```diff
@@ -88,9 +88,13 @@ type TestArtifact struct {
 
 // Validate examines an entire artifact structure and reports if there
 // are any logical inconsistencies with the data.
-func (a *TestArtifact) Validate() error {
+func (a *TestArtifact) Validate(conf BucketConfiguration) error {
```

Comment:
I'd avoid passing conf here, and just add a validate method to the report at the top level to set this default (and maybe validate all the artifacts?) 

There are some methods on the grip.Catcher interface that would help in writing that function.